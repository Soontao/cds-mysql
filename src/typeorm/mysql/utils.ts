
/**
 * equal string without case sensitive
 * 
 * @param s1 first string
 * @param s2 second string
 * @returns true if both two string are equal without case sensitive
 */
export function equalWithoutCase(s1: string, s2: string) {
  if (s1 === undefined && s2 === undefined) {
    return true;
  }
  if (s1?.length === 0 && s2?.length === 0) {
    return true;
  }
  if(s1 === s2) {
    return true;
  }
  if (s1?.length === s2?.length && s1?.toLowerCase() === s2?.toLowerCase()) {
    return true;
  }

  return false;
}