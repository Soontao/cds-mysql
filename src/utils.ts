
/**
 * overwrite cds compiler type mapping
 */
export function overwriteCDSCoreTypes() {

  const { cdsToSqlTypes } = require("@sap/cds-compiler/lib/render/utils/common");

  cdsToSqlTypes.sqlite = {
    "cds.Binary": "CHAR",
    "cds.hana.BINARY": "CHAR",
    "cds.hana.SMALLDECIMAL": "DECIMAL",
  };

}