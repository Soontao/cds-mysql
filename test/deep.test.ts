import { setupTest } from "cds-internal-tool";
import { doAfterAll } from "./utils";

describe("Deep Operation Test Suite", () => {

  const client = setupTest(__dirname, "./resources/deep");
  client.defaults.auth = { username: "alice", password: "admin" };

  afterAll(doAfterAll);

  it("should support get metadata", async () => {
    const response = await client.get("/deep/$metadata");
    expect(response.status).toBe(200);
    expect(response.data).toMatch(/Person/);
  });

  it("should support deep composition creation", async () => {
    const response = await client.post("/deep/Person", {
      ID: 1,
      Name: "Person 1",
      addresses: [
        {
          ID: 1,
          Country: "CN",
          City: "Chengdu",
        },
        {
          ID: 2,
          Country: "CN",
          City: "Shanghai",
        },
        {
          ID: 3,
          Country: "CN",
          City: "Shanghai",
        },
      ]
    });
    expect(response.status).toBe(201);
    expect(response.data).toMatchSnapshot();
  });


});