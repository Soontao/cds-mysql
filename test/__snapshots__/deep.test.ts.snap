// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deep Operation Test Suite should support deep composition creation 1`] = `
Object {
  "@odata.context": "$metadata#Person(addresses())/$entity",
  "ID": 1,
  "Name": "Person 1",
  "addresses": Array [
    Object {
      "City": "Chengdu",
      "Country": "CN",
      "ID": 1,
      "createdAt": "2022-06-15T07:40:23.115Z",
      "createdBy": "alice",
      "modifiedAt": "2022-06-15T07:40:23.115Z",
      "modifiedBy": "alice",
      "up__ID": 1,
    },
    Object {
      "City": "Shanghai",
      "Country": "CN",
      "ID": 2,
      "createdAt": "2022-06-15T07:40:23.115Z",
      "createdBy": "alice",
      "modifiedAt": "2022-06-15T07:40:23.115Z",
      "modifiedBy": "alice",
      "up__ID": 1,
    },
    Object {
      "City": "Shanghai",
      "Country": "CN",
      "ID": 3,
      "createdAt": "2022-06-15T07:40:23.115Z",
      "createdBy": "alice",
      "modifiedAt": "2022-06-15T07:40:23.115Z",
      "modifiedBy": "alice",
      "up__ID": 1,
    },
  ],
  "createdAt": "2022-06-15T07:40:23.115Z",
  "createdBy": "alice",
  "modifiedAt": "2022-06-15T07:40:23.115Z",
  "modifiedBy": "alice",
}
`;
